src/Common/MultiSelectCustomization.tsx

```diff
@@ -0,0 +1,148 @@
+                disabled={props.isDisabled}
+                onChange={(e) => selectOption(data)}
+                type="checkbox"
+                style={{ height: '16px', width: '16px', flex: '0 0 16px' }}
+                className="mr-8"
+            />
+            <components.Option {...props} />
+        </div>
+    )
+}
+
+export const SingleSelectOption = (props) => {
+    const { selectOption, data } = props
+    const style = { height: '16px', width: '16px', flex: '0 0 16px' }
+    const onClick = (e) => selectOption(data)
+    return (
+        <div className="flex left pl-12" style={{ background: props.isFocused ? 'var(--N100)' : 'transparent' }}>
+            {props.isSelected ? (
+                <Check onClick={onClick} className="mr-8 icon-dim-16" style={style} />
+            ) : (
+                <span onClick={onClick} className="mr-8" style={style} />
+            )}
+            <components.Option {...props} />
+        </div>
+    )
+}
+
+export const MultiValueContainer = (props) => {
+    const { children, data, innerProps, selectProps } = props
+    const { label, value } = data
+    return (
+        <components.MultiValueContainer {...{ data, innerProps, selectProps }}>
+            <div className={`flex fs-12 ml-4 cn-9`}>{label}</div>
+            {children[1]}
+        </components.MultiValueContainer>
+    )
+}
+
+export const ClearIndicator = (props) => {
+    const {
+        children = <ClearIcon className="icon-dim-24" />,
+        getStyles,
+        innerProps: { ref, ...restInnerProps },
+    } = props
+    return (
+        <div {...restInnerProps} ref={ref} style={getStyles('clearIndicator', props)}>
+            <div className="flex pointer">{children}</div>
+        </div>
+    )
+}
+
+export const MultiValueRemove = (props) => {
+    const {
+        data,
+        innerProps: { onClick, onMouseDown },
+        selectProps,
+    } = props
+    return (
+        <components.MultiValueRemove {...props}>
+            <ClearIcon
+                {...{ onClick, onMouseDown }}
+                onClick={(e) => onClick(data)}
+                style={{ height: '18px', width: '18px' }}
+            />
+        </components.MultiValueRemove>
+    )
+}
+
+export const MultiValueChipContainer = ({ validator, isAllSelected = false, ...props }) => {
+    const { children, data, innerProps, selectProps } = props
+    const { label, value } = data
+    const isValidEmail = validator ? validator(value) : true
+
+    if (isAllSelected && value !== '*') return null
+
+    return (
+        <components.MultiValueContainer {...{ data, innerProps, selectProps }}>
+            <div className={`flex fs-12 pl-4 pr-4`}>
+                {!isValidEmail && <RedWarning className="mr-4 icon-dim-16" />}
+                <div className={`${isValidEmail ? 'cn-9' : 'cr-5'}`}>{label}</div>
+            </div>
+            {children[1]}
+        </components.MultiValueContainer>
+    )
+}
+
+export const multiSelectStyles = {
```

Comment:
Can we move this and above all functions in utils

Reply:
It is kind of util only